src/FastConv/FastConv.cpp

```diff
@@ -0,0 +1,356 @@
+//        cout << "[0]" << endl;
+//        checkData(m_ppfMulSplitBuffer[0], m_iFftLength/2+1);
+//        cout << "[1]" << endl;
+//        checkData(m_ppfMulSplitBuffer[1], m_iFftLength/2+1);
+//        cout << "[2]" << endl;
+//        checkData(m_ppfMulSplitBuffer[2], m_iFftLength/2+1);
+//        cout << "[3]" << endl;
+//        checkData(m_ppfMulSplitBuffer[3], m_iFftLength/2+1);
+//        cout << "[4]" << endl;
+//        checkData(m_ppfMulSplitBuffer[4], m_iFftLength/2+1);
+//        cout << "block end" << endl;
+    }
+    
+    m_pCFft->mergeRealImag(m_ppfMulBuffer[4], m_ppfMulSplitBuffer[4], m_ppfMulSplitBuffer[5]);
+    
+    m_pCFft->doInvFft(pfMulOut, m_ppfMulBuffer[4]);
+    
+    for (int i = 0; i < m_iFftLength; i++)
+        pfMulOut[i] = pfMulOut[i] * m_iFftLength;
+    
+//    checkData(pfMul1, m_iBlockLength);
+//
+//    checkData(m_ppfMulSplitBuffer[0], m_iFftLength/2+1);
+//
+//    checkData(m_ppfMulSplitBuffer[4], m_iFftLength/2+1);
+//
+//    checkData(pfMulOut, m_iFftLength);
+    
+    return Error_t::kNoError;
+}
+
+Error_t CFastConv::process (float* pfOutputBuffer, const float *pfInputBuffer, int iLengthOfBuffers )
+{
+    
+    if(m_iCompType == 1)
+    {
+        int tailBufferIdx;
+        int tailPosition;
+        int headerBufferIdx;
+        int headerPosition;
+        
+        // tail addition
+        // all ring buffer has length of 2 * m_iBlockLength
+        // the tail stores in the 2nd half of ring buffer
+        tailBufferIdx = m_iCurBlockIdx;
+        for ( int i = 0; i < m_iDivNums; i++)
+        {
+            tailPosition = m_iDivNums - 1 - i;
+            m_ppCRingBuffFft[tailBufferIdx] -> setReadIdx((tailPosition*2+1)*m_iBlockLength);
+            for (int j = 0; j < m_iBlockLength; j++)
+                pfOutputBuffer[j] += m_ppCRingBuffFft[tailBufferIdx] -> getPostInc();
+            tailBufferIdx = (tailBufferIdx+1) % m_iDivNums;
+        }
+        
+        // calculation and store in ringBuffer
+        // data length is m_iFftLength
+        for(int i = 0; i < m_iDivNums; i++)
+        {
+            fftMul(m_pfOutputBufer, pfInputBuffer, m_ppfIRFft[i], m_iBlockLength, m_iBlockLength);
+            m_ppCRingBuffFft[m_iCurBlockIdx] -> put(m_pfOutputBufer, m_iFftLength);
+            m_ppCRingBuffFft[m_iCurBlockIdx] -> setWriteIdx(2*m_iBlockLength*(i+1));
+        }
+        
+        // header addition
+        // the header stores in the 1st half of ring buffer
+        headerBufferIdx = (m_iCurBlockIdx + 1) % m_iDivNums;
+        for ( int i = 0; i < m_iDivNums; i++)
+        {
+            headerPosition = m_iDivNums - 1 - i;
+            m_ppCRingBuffFft[headerBufferIdx] -> setReadIdx(headerPosition*2*m_iBlockLength);
+            for (int j = 0; j < m_iBlockLength; j++)
+                pfOutputBuffer[j] += m_ppCRingBuffFft[headerBufferIdx] -> getPostInc();
+            headerBufferIdx = (headerBufferIdx+1) % m_iDivNums;
+        }
```

Comment:
A problem I see here is every block in output buffer is having a full update every time a new block is filled, which is unnecessary and may cause both confusion and leads to bugs, and also causing performance deficit.